mirrord/kube/src/api/container.rs

```diff
@@ -211,10 +208,11 @@ impl ContainerApi for JobContainer {
                                 }
                             }
                         ],
+                        "imagePullSecrets": agent.image_pull_secrets,
                         "containers": [
                             {
                                 "name": "mirrord-agent",
-                                "image": agent_image,
+                                "image": Self::agent_image(agent),
```

Comment:
Can't you keep the `agent_image` we had outside, and just call `agent_image.clone()` to clone the string value here?

Reply:
Why do I need agent_image outside though?

Reply:
You don't, I find it a bit more readable (but that's debatable).

I'll leave it up to you what looks better.

Reply:
I think we should initialize variables as close as possible to where they're actually used. It's easier to follow when I read through the function (rather than seeing the variable being initialized, going through a bunch of other unrelated code, then seeing how it's used).